server/src/main/java/org/elasticsearch/indices/IndicesQueryCache.java

```diff
@@ -85,34 +85,42 @@ public IndicesQueryCache(Settings settings) {
         sharedRamBytesUsed = 0;
     }
 
-    /** Get usage statistics for the given shard. */
-    public QueryCacheStats getStats(ShardId shard) {
-        final Map<ShardId, QueryCacheStats> stats = new HashMap<>();
-        for (Map.Entry<ShardId, Stats> entry : shardStats.entrySet()) {
-            stats.put(entry.getKey(), entry.getValue().toQueryCacheStats());
+    private static QueryCacheStats toQueryCacheStatsSafe(@Nullable Stats stats) {
+        return stats == null ? new QueryCacheStats() : stats.toQueryCacheStats();
+    }
+
+    private long getShareOfAdditionalRamBytesUsed(double cacheSize) {
+        // We also have some shared ram usage that we try to distribute proportionally to their number of cache entries of each shard.
+
+        if (shardStats.isEmpty()) {
```

Comment:
Is this worth it as an optimization? Seems like just an empty loop below should be good enough?

Reply:
Yeah fair point. I saw a couple of other fast-path possibilities tho, see 630f064.